packages/json-api/src/Registry.ts

```diff
@@ -0,0 +1,92 @@
+import { JSONValue, JSONAPIDocument, JSONAPIResource } from "./jsonApiTypes";
+import {
+  IncludesMap,
+  MapObject,
+  Registry,
+  Resource,
+  ResourceParseOptions,
+  ResourceSpec
+} from "./types";
+import { identity, kebabCaseDeep, camelCaseDeep, TransformFunc } from "./utils";
+import ResourceImpl from "./Resource";
+
+export type RegistryCreateOptions = {
+  keyTransform?: "kebab";
+};
+
+export default class RegistryImpl implements Registry {
+  private options?: RegistryCreateOptions;
+
+  private resources: MapObject<Resource>;
```

Comment:
`Field is assigned only in the constructor and can be made readonly`

Reply:
Good point, fixed.